mne/stats/regression.py

```diff
@@ -246,6 +248,16 @@ def linear_regression_raw(raw, events, event_id=None, tmin=-.1, tmax=1,
     """
 
     if isinstance(solver, string_types):
+        if solver == 'cholesky':
+            try:
+                from sklearn.linear_model import Ridge
+                def solver(X, y):
+                    return Ridge(fit_intercept=False, copy_X=False, alpha=0.,
+                                 solver='cholesky').fit(X, y.T).coef_
```

Comment:
I had originally planned on just doing that, but sklearn also uses safe_sparse_dot (x is sparse here) and it turned out to be more complicated importing that and writing the solver than just importing ridge. Or is safe_sparse_dot not important?






On Sun, Mar 27, 2016 at 12:27 PM -0700, "Alexandre Gramfort" <notifications@github.com> wrote:












In mne/stats/regression.py:
> @@ -246,6 +248,16 @@ def linear_regression_raw(raw, events, event_id=None, tmin=-.1, tmax=1,
>      """
>  
>      if isinstance(solver, string_types):
> +        if solver == 'cholesky':
> +            try:
> +                from sklearn.linear_model import Ridge
> +                def solver(X, y):
> +                    return Ridge(fit_intercept=False, copy_X=False, alpha=0.,
> +                                 solver='cholesky').fit(X, y.T).coef_



you don't need sklearn for this. What it does is just:

linalg.solve(np.dot(X.T, X) , np.dot(X.T, y), sym_pos=True, overwrite_a=True)




—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub